e2e-tests/tests/util.ts

```diff
@@ -186,3 +199,149 @@ export async function extractRevertReason(context: { web3: Web3 }, transactionHa
 		return context.web3.utils.hexToUtf8("0x" + reasonHex.slice(64)).trim(); // skip the padding and remove return carriage
 	}
 }
+
+// Generic function to send a transaction and wait for finalization
+export async function sendTxAndWaitForFinalization(
+	api: ApiPromise,
+	tx: SubmittableExtrinsic<ApiTypes>,
+	signer: KeyringPair,
+	waitNBlocks = 10
+) {
+	return new Promise((resolve, reject) => {
+		try {
+			let blockCount = 0;
```

Comment:
We're passing a callback to `signAndSend` so it's better to explicitly unsubscribe that listener.

```suggestion
			let blockCount = 0;
			let unsubTx: void => ();
```